src/sidebar/services/test/groups-test.js

```diff
@@ -300,6 +358,71 @@ describe('groups', function() {
       });
     });
 
+    it('both groups are in the final groups list when an annotation and a group are linked to', () => {
+      // This can happen if the linked to annotation and group are configured by
+      // the frame embedding the client.
+      const svc = service();
+
+      fakeSettings.group = 'out-of-scope';
+      fakeSettings.annotations = 'ann-id';
+
+      fakeApi.profile.groups.read.returns(Promise.resolve([]));
+      fakeApi.groups.list.returns(
+        Promise.resolve([
+          { name: 'BioPub', id: 'biopub' },
+          { name: 'Public', id: '__world__' },
+        ])
+      );
+      fakeApi.group.read.returns(
+        Promise.resolve({ name: 'Restricted', id: 'out-of-scope' })
+      );
+      fakeApi.annotation.get.returns(
+        Promise.resolve({
+          id: 'ann-id',
+          group: '__world__',
+        })
+      );
+
+      // The user is logged out.
+      fakeAuth.tokenGetter.returns(null);
+
+      return svc.load().then(groups => {
+        const linkedToGroupShown = groups.some(g => g.id === 'out-of-scope');
+        assert.isTrue(linkedToGroupShown);
+        const linkedToAnnGroupShown = groups.some(g => g.id === '__world__');
+        assert.isTrue(linkedToAnnGroupShown);
+      });
+    });
+
+    it('includes the "Public" group if the user links to it', () => {
+      /* Set up the test under conditions that would otherwise
+       * not return the Public group. Aka: the user is logged
+       * out and there are associated groups.
+       */
```

Comment:
Small nit. The convention in our code is to use `//` for inline comments, including multi-line ones and `/**` only for JSDoc comments.